### PR TITLE
This commit adds the ability to use SSL with MySQL.  The docker-entry…

### PR DESCRIPTION
**Pull request by icsy7867**
…point.sh script has been modified to use mysqli and real_connect instead of the mysqli function itself.  This was done so that we can add support for adding a path /etc/ssl/certs/db-ca.crt.  The PHP uses a file_exists function to verify it exists before adding it into the code.  This cert needs to be mounted to the docker container using a volume mount:

	-v /path/to/cert/ca.crt:/etc/ssl/certs/db-ca.crt

Additionally, the yourls code needs an additional piece to tell the PDO driver to use the same cert for verification.  This file was included as db.php and must also be mounted with a volume mount:

        -v /path/to/db.php:/var/www/html/user/db.php

### CODE REVIEW

**Comment by icsy7867** [comment]
Also created a pull request on the main YOURLS code.
https://github.com/YOURLS/YOURLS/pull/3005

If this is implemented, then the db.php file and settings can safely be ignored. (Unless the option to verify a SQL SSL Connection with a CA cert is still desired)

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -40,9 +40,13 @@ if (is_numeric($socket)) {
 	$socket = null;
 }
 
-$maxTries = 10;
+$maxTries = 20;
```
Comment (LeoColomb) [inline docker-entrypoint.sh:43]: What is the reason behind this change?

File: db.php [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<?php
```
Comment (LeoColomb) [inline db.php:1]: Is it worth adding this whole file for all docker deployments?
What is the value compared to a volume mount?

File: docker-entrypoint.sh [hunk 3]
```diff
@@ -40,9 +40,13 @@ if (is_numeric($socket)) {
...
 do {
-	$mysql = new mysqli($host, YOURLS_DB_USER, YOURLS_DB_PASS, '', $port, $socket);
+
+        $mysql = mysqli_init();
+        if( file_exists("/etc/ssl/certs/db-ca.crt") ) { $mysql->ssl_set(NULL, NULL, "/etc/ssl/certs/db-ca.crt", NULL, NULL); }
```
Comment (LeoColomb) [inline docker-entrypoint.sh:47]: `/etc/ssl/certs/db-ca.crt` might or might not exist. Is that path has to be specified?

**Review by LeoColomb — CHANGES_REQUESTED** [review]
Thanks for the pull request @icsy7867!
That there is few points to review first to establish the value of these additions.
Don't forget that the main priority is to keep things simple and not enforce any custom settings, especially on the operating side (MySQL, behaviors, etc.).

File: docker-entrypoint.sh (hunk 1 above)
Comment (icsy7867) [reply docker-entrypoint.sh:43]: This doesn't have to change.  It just gives the container more time alive before crashing and stopping which makes it a little easier to troubleshoot.

Assuming people aren't running into issues it can be set to 10, or even 4

File: db.php (hunk 2 above)
Comment (icsy7867) [reply db.php:1]: More for reference.  The forums had some examples but nothing in detail.  You could just as.easily add a blurb on the docker hub page.  I don't think the actual file will be pulled into the docker container as it's not referenced in the Dockerfile.

File: docker-entrypoint.sh (hunk 3 above)
Comment (icsy7867) [reply docker-entrypoint.sh:47]: The docker container needs a static path, unless you want to implement an environment variable.

The "file_exist" function gives non SSL people functionality.

Otherwise it needs to be documented on the docker hub how to pass your ca.crt file using a docker volume mount.

Comment (LeoColomb) [reply docker-entrypoint.sh:47]: > unless you want to implement an environment variable

Really good suggestion, seems much more appropriate!

File: docker-entrypoint.sh (hunk 1 above)
Comment (LeoColomb) [reply docker-entrypoint.sh:43]: Then I would prefer not to change the value 🙂 

```suggestion
$maxTries = 10;
```

File: db.php (hunk 2 above)
Comment (LeoColomb) [reply db.php:1]: Ah ok, good.
If not inside the Docker image, it should not live inside this repository.
Probably better to suggest a documentation update if so, indeed.

Comment (icsy7867) [reply db.php:1]: Would you like for me to make a stab at a documentation addition?  If so, how would you like to receive that?

100% your project!  Just glad to share my specific use case

File: docker-entrypoint.sh (hunk 3 above)
Comment (icsy7867) [reply docker-entrypoint.sh:47]: The only problem with an environment variable, is that you would need the environment variable for the path as well as a volume mount for the ca.crt.

It would be cleaner, imo, to simply mount a ca.crt to a specific file/directory and check that file to exist (as done in my suggested pull)

But I think either method is fine as long as it's documented.

**Comment by LeoColomb** [comment]
Ok, I reviewed the situation here and I have updated the entrypoint script to remove the mysql part (see #101).
It resolves the case you raised, and does not add clear regression (the database is expected to be created before).

The only thing left to cover 100% of your report is the `db.php` file. As this is always customized for the specific project deployed, I think the initial discussion contains and document its usage properly (https://github.com/YOURLS/YOURLS/discussions/3003). 

For these reasons, I'm closing this pull request.
Thanks again for submitting, really appreciated anyway!
